### ISSUE TITLE
 Error: node stack overflow Error during wrapup: node stack overflow when doing AlignSeqs using DECIPHER  

### ISSUE THREAD

**Issue by ezandi** [benjjneb/dada2#305]
I am doing DNA sequence alignment using DECIPHER (AlignSeqs). I am getting the error in the second step (Clustering into groups by similarity:)

alignmentFls <- AlignSeqs(DNAStringSet(seqFls), processors=NULL, anchor=NA)

Determining distance matrix based on shared 7-mers: |===============================================================================================================================================| 100%

Time difference of 104.46 secs

Clustering into groups by similarity: |===============================================================================================================================================| 100% Error: node stack overflow Error during wrapup: node stack overflow

Does anyone know what the issue is and potential solutions?  This happens when I use 24 samples to generate the seq file. However, when I use a smaller number of samples it works fine!

Thanks,

Ebi

**Comment by benjjneb** [comment]
This error is arising due to another package (DECIPHER) running out of stack space. There is a previous discussion here (https://github.com/benjjneb/dada2/issues/241) and here (https://github.com/spholmes/F1000_workflow/issues/17)

Could you take a look and see if those help? An alternative is to use the `msa` package to do the multiple-alignment step, although the methods they have implemented are slower (https://bioconductor.org/packages/release/bioc/html/msa.html).

**Comment by ezandi** [comment]
In my case removing chimeras did not help!  I have to figure something out or use msa!

Thanks,

Ebi

**Comment by digitalwright** [comment]
This issue has been fixed in DECIPHER version 2.6. Please update your R and then reinstall DECIPHER to begin using the latest version. Sorry for the delay, as I did not see the original post.

**Comment by GeoMicroSoares** [comment]
Sorry in advance to 
@digitalwright trying to do the same thing as @ezandi with AlignSeqs() on R version 3.4.3 (2017-11-30) -- "Kite-Eating Tree" and DECIPHER ‘2.6.0’ with around 70,000 260-400bp sequences. However, R session aborts everytime. Any ideas? Followed the same script as [here](https://github.com/benjjneb/dada2/issues/241).

**Comment by digitalwright** [comment]
If I understand correctly, it sounds like you are trying to align ~70,000 sequences with DECIPHER.  There is currently a limit of 46,340 with the default (automatic) guide tree, and the plan is to double that in the next release (April 2018).  My recommendation would be to cluster the sequences at high-identity prior to alignment so that the number being aligned is much smaller.  You could also try supplying a chained guide tree (refer the DECIPHER vignette about alignment) and set iterations = 0.  I will add a check for the number of sequences in the next version, and report an error if over the limit.

**Comment by GeoMicroSoares** [comment]
Thanks for the quick reply @digitalwright! Wouldn't want to cluster the sequences given that with this alignment I'm aiming at generating a phylogenetic tree to include in a phyloseq object.
Looking forward for that next version!

**Comment by digitalwright** [comment]
Forgot to ask:  have you de-replicated your sequences?  That is, are their ~70,000 unique sequences?

**Comment by GeoMicroSoares** [comment]
They are indeed @digitalwright. Environmental microbiomes...

**Comment by digitalwright** [comment]
@GeoMicroSoares I have completed the version of DECIPHER for aligning more than ~46k sequences.  It can now theoretically scale to a practically unlimited number of sequences.  I have tested it for aligning over 100k sequences and it works well if enough memory is available (tens of GB).  I was happy to see that the alignment was good too!

This update will be released by April, but [send me an email](http://www.wrightlabscience.com/p/contact.html) if you would like to try it out earlier.
